OpenRA.Mods.Common/Traits/Air/Aircraft.cs

```diff
@@ -763,7 +763,10 @@ protected virtual void OnBecomingIdle(Actor self)
 					return;
 				}
 
-				if (Info.IdleBehavior != IdleBehaviorType.Land && dat != Info.CruiseAltitude)
+				// If dat >= CruiseAltitude, the aircraft which should fly in circles in idle
+				// will stop in the mid-air while they are passing terrain with different height.
+				// Make sure the dat < CruiseAltitude when the aircraft queue a TakeOff Activtiy here.
+				if (Info.IdleBehavior != IdleBehaviorType.Land && dat < Info.CruiseAltitude)
```

Comment:
I wonder why is this even needed. Doesn't `FlyIdle` try to maintain CruiseAltitude?

Reply:
it comes from bfcdb3a8a28084bd40ba89cec88d257e473daab3, we need @reaperrr here to have a look, in case we forget something hacky.

Reply:
I don't find any problem on delete the take off here on testing, I will do the change as you said.

Reply:
deleted

Reply:
it seems back then instead of FlyIdle activity we had FlyCircle